invoice_modify_header/invoice.py

```diff
@@ -0,0 +1,125 @@
+from trytond.pool import Pool, PoolMeta
+from trytond.model import Model, ModelView
+from trytond.pyson import Eval
+from trytond.wizard import Button, StateTransition, StateView, Wizard
+from trytond.model.exceptions import AccessError
+from trytond.i18n import gettext
+from trytond.transaction import Transaction
+
+
+class Invoice(metaclass=PoolMeta):
+    __name__ = 'account.invoice'
+
+    @classmethod
+    def __setup__(cls):
+        super().__setup__()
+        # readonly party in case has lines
+        _READONLY = (cls.party.states['readonly']
+            | (Eval('lines', [0]) & Eval('party')))
+        cls.party.states['readonly'] = _READONLY
+
+        cls._buttons.update({
+                'modify_header': {
+                    'invisible': ((Eval('state') != 'draft')
+                        | ~Eval('lines', [-1])),
+                    'depends': ['state'],
+                    },
+                })
+
+    @classmethod
+    def view_attributes(cls):
+        attributes = super().view_attributes()
+        if Transaction().context.get('modify_header'):
+            attributes.extend([
+                    ('//group[@id="states"]', 'states', {'invisible': True}),
+                    ('//group[@id="taxes_amount_state"]', 'states', {'invisible': True}),
+                    ('//group[@id="links"]', 'states', {'invisible': True}),
+                    ('//group[@id="buttons"]', 'states', {'invisible': True}),
+                    ])
+        return attributes
+
+    @classmethod
+    @ModelView.button_action('account_es.wizard_modify_header')
+    def modify_header(cls, invoices):
+        pass
+
+
+class ModifyHeaderStateView(StateView):
+    def get_view(self, wizard, state_name):
+        with Transaction().set_context(modify_header=True):
+            return super(ModifyHeaderStateView, self).get_view(
+                wizard, state_name)
+
+    def get_defaults(self, wizard, state_name, fields):
+        return {}
+
+
+class ModifyHeader(Wizard):
+    "Modify Header"
+    __name__ = 'invoice.modify_header'
+    start = ModifyHeaderStateView('account.invoice',
+        'account_es.modify_header_form', [
+            Button("Cancel", 'end', 'tryton-cancel'),
+            Button("Modify", 'modify', 'tryton-ok', default=True),
+            ])
+    modify = StateTransition()
+
+    def get_invoice(self):
+        if self.record.state != 'draft':
+            # TODO afegir message
```

Comment:
The message.py has the message. ¿?

What's missing is the translations.

Reply:
Done 059d719cb831260f81fb10bdfdde6f4a86998200